### PR TITLE
feat(project): include bindable

### PR DESCRIPTION
**Pull request by bas080**
- when defined it uses these paths to allow filtering on

- when both exclude and include columns is defined, a warning is shown and the
  exclude paths are ignored.

### CODE REVIEW

File: src/filter.js [hunk 1]
```diff
@@ -252,29 +254,60 @@ export class Filter extends CriteriaBuilder {
...
-        continue;
-      }
+    if (includeColumns) {
+      paths = includeColumns;
+    }
 
-      let columnName = (entityName) ? entityName + '.' + column : column;
+    if (!paths) {
+      paths = Object.keys(columns);
+    }
```
Comment (doktordirk) [inline src/filter.js:274]: that.s impossible as i see it

File: src/filter.js [hunk 2]
```diff
@@ -252,29 +254,60 @@ export class Filter extends CriteriaBuilder {
...
 
-    if (this.showIdColumns) {
-      columns.id = 'number';
+    let excludeColumns = splitOnColumnOrIdentity(this.excludeColumns);
+    let includeColumns = splitOnColumnOrIdentity(this.includeColumns);
+
+    if (excludeColumns && includeColumns) {
+      excludeColumns = [];
+
+      logger.warn(`both excludeColumns and includeColumns bindables are defined, ignoring excludeColumns`);
```
Comment (doktordirk) [inline src/filter.js:265]: that seems unneeded. keeping both active does lead to the expected results. hence this only complucate things

File: src/filter.js (hunk 1 above)
Comment (doktordirk) [reply src/filter.js:274]: probably you meant path.length===0

File: src/filter.js (hunk 2 above)
Comment (bas080) [reply src/filter.js:265]: It is a minor optimization and increases explicitness. Furthermore it avoids paths being filtered when include is defined.

File: src/filter.js (hunk 1 above)
Comment (bas080) [reply src/filter.js:274]: Although doing a `![]` results in a true.

Comment (doktordirk) [reply src/filter.js:274]: hm oh. :smile: well, i prefer the change anyways

File: src/filter.js (hunk 2 above)
Comment (doktordirk) [reply src/filter.js:265]: yeah well i see use cases for exactly that when both are generated. can we get a second opinion on that @jeremyvergnas @VMBindraban

Comment (VMBindraban) [reply src/filter.js:265]: You're right, it should be possible to bind both.
If a field is present in `include` and `exclude` it should give priority to `exclude`. (beter hide then show information)

Comment (jeremyvergnas) [reply src/filter.js:265]: I still think `include` should be the priority. So you specify exactly what you want to show.

With `exclude` you might forgot to exclude a column in my opinion.

Comment (doktordirk) [reply src/filter.js:265]: ohje, i would vote for exclude be the priority. it is a stronger statement imho eg sensiivee data or breaking data

Comment (doktordirk) [reply src/filter.js:265]: besides it's much easier. as exclude from the include list does the same as include from what's not excluded

Comment (bas080) [reply src/filter.js:265]: When it comes to sensitive data it's much harder to explicitly expose that data then to implicitly expose it. There is a reason why sails says that you should disable exposing your controller actions in production.

If you put the password property in the include bindable you propably want to show the password. If you forget to put password in exclude you might have made a mistake.

So explicit over implicit. Thus include should get priority.

**Comment by RWOverdijk** [comment]
No need for this. `columns` is essentially the same thing. My fault, sorry.

**Pull request closed without merging**